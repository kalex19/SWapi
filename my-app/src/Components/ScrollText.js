import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ScrollText.scss';

export default class ScrollText extends Component {
	constructor() {
		super();
		this.state = {
			film: {}
		};
	}

	componentDidMount() {
		const range = 7;
		const num = Math.floor(Math.random() * Math.floor(range)) + 1;
		const filmUrl = `https://swapi.co/api/films/${num}`;
		fetch(filmUrl)
			.then(response => response.json())
			.then(film => this.setState({ film }))
			.catch(error => console.log(error));
	}

	render() {
		let scrollText = this.state.film.opening_crawl;
		let movieTitle = this.state.film.title;
		let movieReleaseDate = this.state.film.release_date;

		return (
			<div>
				<p>{scrollText}</p>
				<p>{movieTitle}</p>
				<p>{movieReleaseDate}</p>
			</div>
		);
	}
}

ScrollText.propTypes = {
	film: PropTypes.object
};
