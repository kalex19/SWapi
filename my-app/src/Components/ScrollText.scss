@import '../variables.scss';

.scrollContainer {
  background-color: black;
  padding: 3%;
  margin: 2%;
  animation: slideup 3s linear 1s infinite running;
}

.scrollText {
  text-shadow: 0 0 20px #50ffff, 1px 1px 30px #cea7d0, 2px 2px 30px #4effff, -1px -1px 30px #ad9fca, -2px -2px 30px #feeaac;
  color: $main-shadow;
  animation: bounceIn 2s;
  pointer-events: none;
}

@keyframes bounceIn {
  0% {
    transform: scale(0.1);
    opacity: 0;
  }

  60% {
    transform: scale(1.2);
    opacity: 1;
  }

  100% {
    transform: scale(1);
  }
}