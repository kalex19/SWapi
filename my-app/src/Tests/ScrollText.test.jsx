import React from 'react';
import { shallow } from 'enzyme';
import ScrollText from '../Components/ScrollText';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ScrollText', () => {
	let wrapper;
	let mockFilm;

	beforeEach(() => {
		wrapper = shallow(<ScrollText />);
		mockFilm = {
			name: 'film',
			opening_crawl: 'opening crawl'
		}
	});

	window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
		json: () => Promise.resolve(mockFilm)
	}))

	it('should mock snapshot', async () => {
		await expect(wrapper).toMatchSnapshot();
	});

	it('should have default state', async () => {
		await expect(wrapper.state()).toEqual({
			film: {}
		});
	});

	it('calls fetch with the correct data', async () => {
		const url = 'swapi.co/api/films'
		const options = {
		method: 'POST',
		body: JSON.stringify({ film: mockFilm}),
		headers: {
			'Content-Type': 'application/json'
		}
		}

 	 	wrapper.setState({film: mockFilm})
  	await expect(window.fetch).toHaveBeenCalledWith(url, options)
  });

  it('sets the error in state if the fetch fails', async () => {
			window.fetch = jest.fn().mockImplementationOnce(() => Promise.reject(
				new Error('Fetch failed')
			)
				await expect(wrapper.state('errorStatus')).toEqual('Error adding grocery')
			})
	})
  });

});
