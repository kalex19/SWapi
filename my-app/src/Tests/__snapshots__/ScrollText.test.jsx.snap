// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText should mock snapshot 1`] = `undefined`;
